test/test-barrier.c

```diff
@@ -139,3 +139,11 @@ TEST_IMPL(barrier_serial_thread) {
   return 0;
 }
 
+/* Single thread uv_barrier_wait should return correct return value. */
+TEST_IMPL(barrier_serial_thread_single) {
+  uv_barrier_t barrier;
+
+  ASSERT(0 == uv_barrier_init(&barrier, 1));
+  ASSERT(0 < uv_barrier_wait(&barrier));
```

Comment:
Can you also call `uv_barrier_destroy()`?

Reply:
Done.